Scripts/modelsystem.py

```diff
@@ -292,13 +292,16 @@ def run_iteration(self, previous_iter_impedance, iteration=None):
             mode_shares[mode] = trip_sum[mode].sum() / sum_all.sum()
         self.mode_share.append(mode_shares)
 
+        # Add vans and save demand matrices
+        for ap in self.ass_model.assignment_periods:
+            self.dtm.add_vans(ap.name, self.zdata_forecast.nr_zones)
+            if iteration=="last":
+                self._save_demand_to_omx(ap.name)
+
         # Calculate and return traffic impedance
         for ap in self.ass_model.assignment_periods:
             tp = ap.name
             log.info("Assigning period " + tp)
```

Comment:
I am not sure if I understand correctly but it looks to me that these two lines are the only ones that can fail between `master` and this PR. So is this just a refactoring (= saving all demand matrices before printing "Assigning period xxx") and not really bugfixing?

Reply:
Well, this can fail (badly): https://github.com/HSLdevcom/helmet-model-system/blob/76fdd8c158a1dc543b343250ca5bc3d541476c39/Scripts/modelsystem.py#L305

And because it runs in a loop, having demand matrix saving in the same loop means it will then not be run for consecutive iterations.

Reply:
Ah, I see! If line 305 fails during "aht", there would not be demand matrices saved for "iht" and "pt" - but with this PR, there would be. Good work!